server/server.go

```diff
@@ -335,67 +321,58 @@ func (s *server) Start(ctx context.Context, req *rpcpb.StartRequest) (*rpcpb.Sta
 		snapshotsDir:        s.cfg.SnapshotsDir,
 	})
 	if err != nil {
-		s.network = nil
 		return nil, err
 	}
 
-	if err := s.network.start(); err != nil {
+	s.log.Info("starting",
+		zap.String("exec-path", execPath),
+		zap.Uint32("num-nodes", numNodes),
+		zap.String("track-subnets", trackSubnets),
+		zap.Int32("pid", pid),
+		zap.String("root-data-dir", rootDataDir),
+		zap.String("plugin-dir", pluginDir),
+		zap.Any("chain-configs", req.ChainConfigs),
+		zap.String("global-node-config", globalNodeConfig),
+	)
+
+	if err := s.network.Start(); err != nil {
 		s.log.Warn("start failed to complete", zap.Error(err))
-		s.network = nil
+		s.stopAndRemoveNetwork()
 		return nil, err
 	}
 
-	// start non-blocking to install local cluster + custom chains (if applicable)
-	// the user is expected to poll cluster status
-	readyCh := make(chan struct{})
-	go s.network.startWait(ctx, chainSpecs, readyCh)
-
 	// update cluster info non-blocking
 	// the user is expected to poll this latest information
 	// to decide cluster/subnet readiness
 	go func() {
-		s.waitChAndUpdateClusterInfo("local cluster", readyCh, false)
-		s.waitChAndUpdateClusterInfo("custom chains", readyCh, true)
+		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
+		defer cancel()
+		err := s.network.CreateChains(ctx, chainSpecs)
```

Comment:
we want this error to be returned by WaitForHealthy

Reply:
https://github.com/ava-labs/avalanche-network-runner/pull/447#discussion_r1120195954